invenio_records_rest/views.py

```diff
@@ -896,8 +897,15 @@ def put(self, pid, record, **kwargs):
         self.check_etag(str(record.revision_id), weak=True)
 
         record.clear()
-        record.update(data)
-        record.commit()
+        try:
+            record.update(data)
+            record.commit()
+        except ValidationError:
+            record_schema = get_schema_name_from_uri(record.get('$schema'))
+            if not record_schema:
+                raise
+            errors = get_validation_errors_for_schema(data, record_schema)
```

Comment:
* actually, this is a bit silly, we're validating records twice now when there's a validation error, which can be expensive for large records. Maybe we should modify the record API to return all validation errors and avoid validating twice?
* what about `POST`?

Reply:
* I am not sure if that would be better. We use [DraftValidator](https://github.com/inspirehep/inspire-schemas/blob/98e0fd3c03fb33eecf6a71b8c32f0b12573b82c9/inspire_schemas/utils.py#L777) to get the errors so I believe that we'll have to use it even if we'll move getting errors to `validate`.
* Do we post on record-editor? In other places we don't need those errors.

Reply:
* We're calling https://github.com/Julian/jsonschema/blob/d16713a4296663f3d62c50b9f9a2893cb380b7af/jsonschema/validators.py#L487 which in turns calls https://github.com/Julian/jsonschema/blob/d16713a4296663f3d62c50b9f9a2893cb380b7af/jsonschema/validators.py#L128-L130 which ends up calling `iter_errors()` too, and raises the first error there. So we do call `iter_errors()` twice, One easy option I see to avoid validating twice is to replace https://github.com/inspirehep/inspirehep/blob/ab59746c15ed1dd852e9d052bc721f2cdba48589/backend/inspirehep/records/api/base.py#L67-L68 by something like:
```
def validate(self):
    self._validation_errors = list(get_validation_errors(self))  # the current API in the inspire-schemas is bad, you need something that infers the schema
    if self._validation_errors:
        raise self._validation_errors[0]
```
then reuse `self._validation_errors` instead of calling `get_validations_errors` again.
* I don't think we post on record-editor, so indeed, we don't need it there for now.